src/pages/admin/student/list/components/studentTable.tsx

```diff
@@ -1,24 +1,67 @@
 import * as React from 'react';
 import { StudentSummary } from '../../../../../model/studentSummary';
 import { StudentRowComponent } from './studentRow';
+import { AutoSizer, Table, Column } from 'react-virtualized';
+const classNames: any = require('./studentTableStyles.scss');
 
-interface IProps {
+interface StudentTableComponentProps {
   studentList: StudentSummary[];
 }
 
-export const StudentTableComponent = (props: IProps = {studentList: []}) => {
-  return (
-    <table>
-      <tbody>
-        {
-          props.studentList.map((student) =>
-            <StudentRowComponent
-              key={student.id}
-              student={student}
-            />,
-          )
-        }
-      </tbody>
-    </table>
-  );
+// More info about react-virtualized:
+// https://github.com/Lemoncode/react-training-ts/tree/master/02%20Sample%20App/04%20TrainingListB
+export const StudentTableComponent: React.StatelessComponent<StudentTableComponentProps> =
+  ({ studentList }) => {
+    return (
+      <AutoSizer disableHeight={true}>
+        {({ width }) => <InnerStudentTableComponent width={width} studentList={studentList} />}
+      </AutoSizer>
+    );
+  };
+
+const rowGetter = (studentList) => ({ index }) => studentList[index];
+const getWidthByPercentage = (width, percentage) => {
+  return (percentage * width) / 100;
 };
+
+interface InnerStudentTableComponentProps extends StudentTableComponentProps {
+  width: number;
+}
+
+export const InnerStudentTableComponent: React.StatelessComponent<InnerStudentTableComponentProps> =
```

Comment:
Finally, after talk with you these days, I think that the better way to test this one is move `Autosizer` stuff outside this component and test only `Table` stuff.